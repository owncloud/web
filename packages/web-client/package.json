{
  "name": "@ownclouders/web-client",
  "version": "11.3.1",
  "description": "ownCloud web client",
  "license": "AGPL-3.0",
  "private": false,
  "type": "module",
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-client",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-client"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "default": "./dist/web-client.js",
        "require": "./dist/web-client.cjs",
        "types": "./dist/src/index.d.ts"
      },
      "./graph": {
        "default": "./dist/web-client/graph.js",
        "require": "./dist/web-client/graph.cjs",
        "types": "./dist/src/graph/index.d.ts"
      },
      "./graph/generated": {
        "default": "./dist/web-client/graph/generated.js",
        "require": "./dist/web-client/graph/generated.cjs",
        "types": "./dist/src/graph/generated/index.d.ts"
      },
      "./ocs": {
        "default": "./dist/web-client/ocs.js",
        "require": "./dist/web-client/ocs.cjs",
        "types": "./dist/src/ocs/index.d.ts"
      },
      "./sse": {
        "default": "./dist/web-client/sse.js",
        "require": "./dist/web-client/sse.cjs",
        "types": "./dist/src/sse/index.d.ts"
      },
      "./webdav": {
        "default": "./dist/web-client/webdav.js",
        "require": "./dist/web-client/webdav.cjs",
        "types": "./dist/src/webdav/index.d.ts"
      }
    }
  },
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./graph": {
      "default": "./src/graph/index.ts"
    },
    "./graph/generated": {
      "default": "./src/graph/generated/index.ts"
    },
    "./ocs": {
      "default": "./src/ocs/index.ts"
    },
    "./sse": {
      "default": "./src/sse/index.ts"
    },
    "./webdav": {
      "default": "./src/webdav/index.ts"
    },
    "./l10n": {
      "default": "./l10n/translations.json"
    }
  },
  "scripts": {
    "generate-openapi": "rm -rf src/graph/generated && docker run --rm -v \"${PWD}/src/graph:/local\" openapitools/openapi-generator-cli generate -i https://raw.githubusercontent.com/owncloud/libre-graph-api/main/api/openapi-spec/v1.0.yaml  -g typescript-axios -o /local/generated",
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && rm -rf package/dist/tests && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@microsoft/fetch-event-source": "^2.0.1",
    "@ownclouders/web-pkg": "workspace:*",
    "axios": "^1.7.7",
    "fast-xml-parser": "^4.5.0",
    "lodash-es": "^4.17.21",
    "luxon": "^3.5.0",
    "uuid": "^11.0.0",
    "webdav": "^5.7.1",
    "xml-js": "^1.6.11",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@types/luxon": "3.4.2",
    "@vitejs/plugin-vue": "5.1.4",
    "clean-publish": "5.0.0",
    "vite-plugin-dts": "4.2.4",
    "vite-plugin-node-polyfills": "0.22.0",
    "vite": "5.4.8"
  }
}
