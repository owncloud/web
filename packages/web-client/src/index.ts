import { graph } from './graph'
import { ocs } from './ocs'
import { webdav } from './webdav'

export * from './errors'
export * from './helpers'
export * from './utils'
export * from './constants'

export { graph, ocs, webdav }
