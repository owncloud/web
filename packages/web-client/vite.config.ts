import { join, resolve } from 'path'
import { defineConfig, searchForWorkspaceRoot } from 'vite'
import dts from 'vite-plugin-dts'
import { nodePolyfills } from 'vite-plugin-node-polyfills'

const projectRootDir = searchForWorkspaceRoot(process.cwd())

export default defineConfig({
  resolve: {
    alias: {
      crypto: join(projectRootDir, 'polyfills/crypto.js')
    }
  },
  build: {
    lib: {
      entry: {
        'web-client': resolve(__dirname, 'src/index.ts'),
        'web-client/graph': resolve(__dirname, 'src/graph/index.ts'),
        'web-client/graph/generated': resolve(__dirname, 'src/graph/generated/index.ts'),
        'web-client/ocs': resolve(__dirname, 'src/ocs/index.ts'),
        'web-client/sse': resolve(__dirname, 'src/sse/index.ts'),
        'web-client/webdav': resolve(__dirname, 'src/webdav/index.ts')
      }
    }
  },
  plugins: [
    nodePolyfills({
      exclude: ['crypto']
    }),
    dts({
      include: ['src'],
      outDir: 'dist/types',
      insertTypesEntry: true
    })
  ]
})
