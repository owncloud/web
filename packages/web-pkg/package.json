{
  "name": "@ownclouders/web-pkg",
  "version": "11.1.3",
  "description": "ownCloud web pkg",
  "license": "AGPL-3.0",
  "main": "src/index.ts",
  "types": "types.d.ts",
  "private": false,
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "type": "module",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-pkg",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-pkg"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "import": "./dist/web-pkg.js",
        "require": "./dist/web-pkg.umd.cjs",
        "types": "./dist/src/index.d.ts"
      }
    }
  },
  "scripts": {
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && rm -rf package/dist/tests package/dist/*.css && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@casl/vue": "^2.2.2",
    "@microsoft/fetch-event-source": "^2.0.1",
    "@ownclouders/design-system": "workspace:^",
    "@ownclouders/web-client": "workspace:^",
    "@sentry/vue": "^8.33.1",
    "@uppy/core": "^3.3.0",
    "@uppy/drop-target": "^2.0.0",
    "@uppy/tus": "^3.1.0",
    "@uppy/utils": "^5.3.0",
    "@uppy/xhr-upload": "^3.0.1",
    "@vavt/cm-extension": "^1.6.0",
    "@vue/shared": "^3.5.11",
    "@vueuse/core": "^11.0.0",
    "axios": "^1.7.7",
    "deepmerge": "^4.2.2",
    "dompurify": "^3.2.3",
    "filesize": "^10.1.0",
    "fuse.js": "^7.0.0",
    "js-generate-password": "^1.0.0",
    "lodash-es": "^4.17.21",
    "luxon": "^3.5.0",
    "mark.js": "^8.11.1",
    "md-editor-v3": "^5.2.1",
    "oidc-client-ts": "^2.4.0 || ^3.0.0",
    "p-queue": "^8.0.0",
    "password-sheriff": "^1.1.1",
    "pinia": "^2.2.4",
    "portal-vue": "^3.0.0",
    "prismjs": "^1.29.0",
    "qs": "^6.13.0",
    "semver": "^7.6.3",
    "uuid": "^11.0.0",
    "vue-concurrency": "^5.0.1",
    "vue-router": "^4.2.5",
    "vue3-gettext": "^2.4.0",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@ownclouders/web-test-helpers": "workspace:^",
    "@types/dompurify": "3.0.5",
    "@types/lodash-es": "4.17.12",
    "@vitest/web-worker": "2.1.2",
    "clean-publish": "5.0.0",
    "vite-plugin-dts": "4.2.4",
    "vite-plugin-node-polyfills": "0.22.0"
  }
}
