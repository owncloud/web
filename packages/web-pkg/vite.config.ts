import { join, resolve } from 'path'
import { defineConfig, searchForWorkspaceRoot } from 'vite'
import dts from 'vite-plugin-dts'
import { nodePolyfills } from 'vite-plugin-node-polyfills'
import vue from '@vitejs/plugin-vue'
import pkg from './package.json' assert { type: 'json' }

const projectRootDir = searchForWorkspaceRoot(process.cwd())
const external = [...Object.keys(pkg.dependencies)]

export default defineConfig({
  resolve: {
    alias: {
      crypto: join(projectRootDir, 'polyfills/crypto.js')
    }
  },
  css: {
    preprocessorOptions: {
      scss: {
        additionalData: `
          @use "sass:math";
          @import "${projectRootDir}/packages/design-system/src/styles/styles";
        `,
        silenceDeprecations: ['legacy-js-api', 'import']
      }
    }
  },
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'web-pkg',
      fileName: 'web-pkg'
    },
    rollupOptions: {
      external
    }
  },
  plugins: [
    vue(),
    nodePolyfills({
      exclude: ['crypto']
    }),
    dts({ exclude: ['**/tests'], include: ['src'], outDir: 'dist/types', insertTypesEntry: true }),
    {
      name: '@ownclouders/vite-plugin-docs',
      transform(src, id) {
        if (id.includes('type=docs')) {
          return {
            code: 'export default {}',
            map: null
          }
        }
      }
    }
  ]
})
