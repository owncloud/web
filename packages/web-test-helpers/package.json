{
  "name": "@ownclouders/web-test-helpers",
  "version": "11.3.2",
  "description": "ownCloud web test helpers",
  "license": "AGPL-3.0",
  "private": false,
  "type": "module",
  "main": "src/index.ts",
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-test-helpers",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-test-helpers"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "import": "./dist/web-test-helpers.es.js",
        "require": "./dist/web-test-helpers.umd.cjs",
        "types": "./dist/src/index.d.ts"
      }
    }
  },
  "scripts": {
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "peerDependencies": {
    "@ownclouders/web-pkg": "workspace:^",
    "@vue/test-utils": "^2.4.6",
    "vue": "^3.5.10"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "5.1.4",
    "clean-publish": "^5.0.0",
    "vite-plugin-dts": "^4.2.3",
    "vite": "^5.4.8"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@casl/vue": "^2.2.2",
    "@ownclouders/design-system": "workspace:^",
    "@ownclouders/web-client": "workspace:^",
    "@pinia/testing": "^0.1.3",
    "axios": "1.7.7",
    "pinia": "2.2.6",
    "vitest-mock-extended": "2.0.2",
    "vue-router": "4.2.5",
    "vue3-gettext": "2.4.0"
  }
}
