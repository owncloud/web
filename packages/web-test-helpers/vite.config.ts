import { resolve } from 'path'
import { defineConfig } from 'vite'
import dts from 'vite-plugin-dts'
import pkg from './package.json' assert { type: 'json' }
import vue from '@vitejs/plugin-vue'

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'web-test-helpers',
      fileName: (format) => `web-test-helpers.${format}.js`
    },
    rollupOptions: {
      external: [...Object.keys(pkg.dependencies), ...Object.keys(pkg.peerDependencies)]
    }
  },
  plugins: [vue(), dts({ include: ['src'], outDir: 'dist/types', insertTypesEntry: true })]
})
