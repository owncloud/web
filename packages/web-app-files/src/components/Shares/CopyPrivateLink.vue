<template>
  <div class="oc-flex oc-flex-middle copy-private-link">
    <oc-button v-oc-tooltip="tooltip" gap-size="none" appearance="raw" @click="copyLinkToClipboard">
      <oc-icon size="small" :name="copied ? 'checkbox-circle' : 'file-copy'" fill-type="line" />
      <span class="oc-ml-xs" v-text="$gettext('Permanent link')"
    /></oc-button>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, PropType } from 'vue'
import { Resource } from '@ownclouders/web-client'
import { useGettext } from 'vue3-gettext'
import { useClipboard } from '@vueuse/core'
import { useMessages } from '@ownclouders/web-pkg'

export default defineComponent({
  props: {
    resource: {
      type: Object as PropType<Resource>,
      required: true
    }
  },
  setup(props) {
    const { $gettext } = useGettext()
    const { showMessage } = useMessages()
    const { copy, copied } = useClipboard({ legacy: true, copiedDuring: 550 })

    const copyLinkToClipboard = () => {
      copy(props.resource.privateLink)
      showMessage({
        title: $gettext('Permanent link copied'),
        desc: $gettext('The permanent link has been copied to your clipboard.')
      })
    }

    const tooltip = computed(() => {
      return $gettext(
        'Copy the link to point your team to this item. Works only for people with existing access.'
      )
    })

    return { copied, tooltip, copyLinkToClipboard }
  }
})
</script>
