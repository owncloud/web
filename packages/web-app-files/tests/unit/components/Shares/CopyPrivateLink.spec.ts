import { mock } from 'vitest-mock-extended'
import { Resource } from '@ownclouders/web-client'
import { defaultPlugins, mount } from 'web-test-helpers'
import CopyPrivateLink from 'web-app-files/src/components/Shares/CopyPrivateLink.vue'
import { useMessages } from '@ownclouders/web-pkg'

const resource = mock<Resource>({
  type: 'folder',
  owner: {
    id: 'marie',
    displayName: 'Marie'
  },
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  size: '740',
  name: 'lorem.txt',
  privateLink: 'https://example.com/fake-private-link'
})

// @vitest-environment jsdom
describe('CopyPrivateLink', () => {
  beforeEach(() => {
    vi.useFakeTimers()
  })

  afterEach(() => {
    vi.useRealTimers()
  })

  it('should render a button', () => {
    const { wrapper } = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('upon clicking it should copy the private link to the clipboard button, render a success message and change icon for half a second', async () => {
    Object.assign(window.navigator, {
      clipboard: {
        writeText: vi.fn().mockImplementation(() => Promise.resolve())
      }
    })

    const { wrapper } = getWrapper()
    const { showMessage } = useMessages()
    expect(showMessage).not.toHaveBeenCalled()

    await wrapper.find('button').trigger('click')
    expect(window.navigator.clipboard.writeText).toHaveBeenCalledWith(resource.privateLink)
    expect(showMessage).toHaveBeenCalledTimes(1)
  })
})

function getWrapper() {
  return {
    wrapper: mount(CopyPrivateLink, {
      props: {
        resource
      },
      global: {
        plugins: [...defaultPlugins()]
      }
    })
  }
}
