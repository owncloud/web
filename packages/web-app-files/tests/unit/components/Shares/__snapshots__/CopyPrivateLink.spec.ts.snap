// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CopyPrivateLink > should render a button 1`] = `
"<div class="oc-flex oc-flex-middle copy-private-link"><button type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-undefined oc-button-passive oc-button-passive-raw">
    <!--v-if-->
    <!-- @slot Content of the button --> <span class="oc-icon oc-icon-s oc-icon-passive"><!----></span> <span class="oc-ml-xs">Permanent link</span>
  </button></div>"
`;
